### PR TITLE
Fix several metadata methods and tests

### PR DESCRIPTION
**Pull request by pmishchenko-ua**
`CHARACTER_MAXIMUM_LENGTH` for TEXT types should be the same as `CHARACTER_OCTET_LENGTH`, as characters can be 1-byte.

### CODE REVIEW

File: ma_helper.c [hunk 1]
```diff
@@ -447,7 +480,7 @@ char *MADB_GetDefaultColumnValue(MYSQL_RES *res, const char *Column)
   return NULL;
 }
 
-SQLLEN MADB_GetDataSize(SQLSMALLINT SqlType, SQLLEN OctetLength, BOOL Unsigned,
+SQLLEN MADB_GetDataSize(SQLSMALLINT SqlType, SQLLEN OctetLength, BOOL isSigned,
```
Comment (AdalbertMemSQL) [inline ma_helper.c:483]: lol
they used a very bad name

File: ma_helper.c [hunk 2]
```diff
@@ -535,40 +573,53 @@ size_t MADB_GetDisplaySize(MYSQL_FIELD *Field, MARIADB_CHARSET_INFO *charset)
...
     {
-      return Field->length/charset->char_maxlen;
+      return MIN(Field->length/charset->char_maxlen, S2_1_GB);
     }
   }
+  // no SingleStore type falls to the group below
+  case MYSQL_TYPE_ENUM:
+  case MYSQL_TYPE_SET:
+  case MYSQL_TYPE_GEOMETRY:
+  case MYSQL_TYPE_VARCHAR:
```
Comment (AdalbertMemSQL) [inline ma_helper.c:604]: Don't we have varchar?

File: ma_helper.c [hunk 3]
```diff
@@ -535,40 +573,53 @@ size_t MADB_GetDisplaySize(MYSQL_FIELD *Field, MARIADB_CHARSET_INFO *charset)
...
     else
     {
-      return Field->length/charset->char_maxlen;
+      return MIN(Field->length/charset->char_maxlen, S2_1_GB);
     }
   }
+  // no SingleStore type falls to the group below
+  case MYSQL_TYPE_ENUM:
+  case MYSQL_TYPE_SET:
+  case MYSQL_TYPE_GEOMETRY:
```
Comment (AdalbertMemSQL) [inline ma_helper.c:603]: Is it correct that all these types are converted to strings?

File: ma_helper.c [hunk 4]
```diff
@@ -535,40 +573,53 @@ size_t MADB_GetDisplaySize(MYSQL_FIELD *Field, MARIADB_CHARSET_INFO *charset)
...
 
 /* {{{ MADB_GetOctetLength */
-size_t MADB_GetOctetLength(MYSQL_FIELD *Field, unsigned short MaxCharLen)
+/**
+  Get the transfer octet length of a field, as defined at:
+    http://msdn2.microsoft.com/en-us/library/ms713979.aspx
+
+  @param[in] Field
+
+  @return  The transfer octet length. Cap at INT_MAX32 due to signed value
```
Comment (AdalbertMemSQL) [inline ma_helper.c:618]: Cap at 1 GB?

File: ma_statement.c [hunk 5]
```diff
@@ -5024,18 +5038,35 @@ SQLRETURN MADB_StmtColumnsNoInfoSchema(MADB_Stmt *Stmt,
...
+      if (field->charsetnr != BINARY_CHARSETNR )
+      {
+        charset = mariadb_get_charset_by_nr(field->charsetnr);
+        char_size = charset ? charset->char_maxlen : 4;
+      }
+      precision = field->length - test((field->flags & UNSIGNED_FLAG) == 0) - test(field->decimals != 0);
+      is_signed = !(field->flags & UNSIGNED_FLAG);
+      if ((column_char_length = MADB_GetDataSize(concise_data_type, column_length, is_signed, precision, field->decimals, char_size)) != SQL_NO_TOTAL)
+      {
+        printf("column_char_length from MADB_GetDataSize is %d, odbc_data_type %d, name %s\n", column_char_length, odbc_data_type, field->name);
```
Comment (AdalbertMemSQL) [inline ma_statement.c:5060]: Hmm... Will this be printed each time?

File: ma_statement.h [hunk 6]
```diff
@@ -361,10 +371,12 @@ void         MADB_CspsFreeDAE(MADB_Stmt *Stmt);
 #define MADB_PROCEDURE_COLUMNSp3 \
   " AS SQL_DATA_TYPE, "\
   MADB_COLUMNS_DATETIME_SUB " AS SQL_DATETIME_SUB, "\
-  "CHARACTER_OCTET_LENGTH :> INT AS CHAR_OCTET_LENGTH, "\
+  "LEAST(CHARACTER_OCTET_LENGTH, 1073741823) AS CHAR_OCTET_LENGTH, "\
   "ORDINAL_POSITION, "\
   "IF(DTD_IDENTIFIER LIKE '%%NOT NULL%%', 'NO', 'YES') AS IS_NULLABLE FROM INFORMATION_SCHEMA.PARAMETERS "
 
+// max_packet_size can't be larger that 1 GB, so 1073741823 is the maximum number of bytes
+// that can be stored in a SingleStore field
```
Comment (AdalbertMemSQL) [inline ma_statement.h:379]: Looks like this comment is added in a wrong place

File: ma_statement.h [hunk 7]
```diff
@@ -324,8 +336,6 @@ void         MADB_CspsFreeDAE(MADB_Stmt *Stmt);
     " WHEN 'mediumint' THEN 0" \
     " WHEN 'int' THEN 0" \
     " WHEN 'bigint' THEN 0" \
-    " WHEN 'double' THEN IF(NUMERIC_SCALE IS NULL, 6, NUMERIC_SCALE)" \
-    " WHEN 'float' THEN IF(NUMERIC_SCALE IS NULL, 0, NUMERIC_SCALE)" \
```
Comment (AdalbertMemSQL) [inline ma_statement.h:328]: Don't we want this query to handle double/float?

File: ma_helper.c (hunk 2 above)
Comment (pmishchenko-ua) [reply ma_helper.c:604]: we do, but it is reported as MYSQL_TYPE_VAR_STRING.

File: ma_statement.h (hunk 7 above)
Comment (pmishchenko-ua) [reply ma_statement.h:328]: If I understand correctly https://learn.microsoft.com/en-us/sql/odbc/reference/appendixes/decimal-digits?view=sql-server-ver16, for double/float ( approximate data types) decimal digits is undefined.

**Review by AdalbertMemSQL — APPROVED** [review]

**Pull request merged**